### PR TITLE
[DOC] Improve and fix comparation and regeneration for golden files

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

- Extract common assertion method for verifying file contents
- Ensure integrity of the file by comparing the line count
- Correct the script name for Spark engine KDF doc generation from `gen_kdf.sh` to `gen_spark_kdf_docs.sh`
- Add `gen_hive_kdf_docs.sh` script for Hive engine KDF doc generation
- Fix incorrect hints for Ranger spec file generation
- shows the line number of the incorrect file content
- Streamingly read file content by line with buffered support
- Regeneration hints:

image

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_

No.

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]

**Comment by bowenliang123** [comment]
May I backport this PR to branch-1.8 for consistency and maintainability? cc @pan3793

File: dev/gen/gen_hive_kdf_docs.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Golden result file:
+# docs/extensions/engines/hive/functions.md
```
Comment (cxzl25) [inline dev/gen/gen_hive_kdf_docs.sh:20]: We also support Flink function `docs/extensions/engines/flink/functions.md`

Comment (bowenliang123) [reply dev/gen/gen_hive_kdf_docs.sh:20]: Which test suite responsible for regenerating this doc?

**Comment by pan3793** [comment]
@bowenliang123 it's fine to have in 1.8

**Comment by bowenliang123** [comment]
Thanks, merged to master (1.9.0) and branch-1.8 (1.8.0).

**Pull request closed without merging**